src/JustSaying/AwsTools/MessageHandling/MessageHandlerWrapper.cs

```diff
@@ -49,14 +48,19 @@ internal sealed class MessageHandlerWrapper
                 throw new Exception("IMessageLock is null. You need to specify an implementation for IMessageLock.");
             }
 
+            if (uniqueKeySelector == null)
```

Comment:
Should throw this in the caller before we start to do any work, rather than in the delegate further along the call path.